apps/web/src/views/GaugesVoting/components/Table/VoteTable/hooks/useGaugeRows.ts

```diff
@@ -62,7 +62,7 @@ const useSelectRowsWithQuery = (gauges: Gauge[] | undefined) => {
       const urlHashes = queryHashes.filter((hash) => !newHashes.includes(hash))
       setSelectRowsHash(urlHashes.concat(newHashes))
     }
-  }, [isLoading, prevVotedGauges, queryHashes])
+  }, [isLoading, prevVotedGauges.map((x) => x.hash).join('-'), queryHashes])
```

Comment:
Does this work?

useEffect(() => {
  if (prevVotedGauges && !isLoading) {
    const prevHashes = prevVotedGauges.map((gauge) => gauge.hash);
    const filteredHashes = queryHashes.filter((hash) => !prevHashes.includes(hash));
    const combinedHashes = [...filteredHashes, ...prevHashes];
    setSelectRowsHash(combinedHashes);
  }
}, [isLoading, prevVotedGauges, queryHashes]);

Reply:
done